src/utransport.rs

```diff
@@ -85,5 +83,327 @@ pub trait UTransport {
     /// # Errors
     ///
     /// Returns an error if the listener could not be unregistered, for example if the given listener does not exist.
-    async fn unregister_listener(&self, topic: UUri, listener: &str) -> Result<(), UStatus>;
+    async fn unregister_listener<T>(&self, topic: UUri, listener: Arc<T>) -> Result<(), UStatus>
+    where
+        T: UListener;
+}
+
+/// A wrapper type around UListener that can be used by `up-client-foo-rust` UPClient libraries
+/// to ease some common development scenarios when we want to compare `UListener`
+///
+/// # Note
+///
+/// Not necessary for end-user uEs to use. Primarily intended for `up-client-foo-rust` UPClient libraries
+///
+/// # Rationale
+///
+/// The wrapper type is implemented such that it can be used in any location you may wish to
+/// hold a generic `UListener`
+///
+/// Also implements necessary traits to allow hashing, so that you may hold the wrapper type in
+/// collections which require that, such as a `HashMap` or `HashSet`
+///
+/// Note that the implementation is such that if the same instance of `Arc<T>` is used multiple times
+/// to construct multiple different wrappers, then these wrappers are all considered equivalent
+/// due to using the Arc's pointer as unique identifier
+///
+/// In the reverse, if two different instances of `Arc<T>` are used to construct two wrappers,
+/// they will not be considered equivalent due to the Arc pointer being different for each instance
```

Comment:
this also needs to be adapted according to the *pointer equality*. Maybe you should simply refer to the `PartialEq::eq` implementation and document there, how Arc<T>s are being compared ...

Reply:
Some good feedback here. I'll try to document this more clearly :+1: